src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/features/SchedulerConfigTest.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.reform.blobrouter.features;
+
+import net.javacrumbs.shedlock.core.LockConfiguration;
+import net.javacrumbs.shedlock.core.LockProvider;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.Profile;
+import org.springframework.test.context.TestPropertySource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest
+@TestPropertySource(
+    properties = {
+        "scheduling.task.delete-dispatched-files.enabled=true",
+        "scheduling.task.delete-dispatched-files.cron: */1 * * * * *"
+    }
+)
+@Profile("integration-test")
+public class SchedulerConfigTest {
+
+    @SpyBean
+    private LockProvider lockProvider;
+
+    @Test
+    public void should_integrate_with_shedlock() throws Exception {
+        ArgumentCaptor<LockConfiguration> configCaptor = ArgumentCaptor.forClass(LockConfiguration.class);
+        Thread.sleep(2000);
```

Comment:
why do we have sleep here @apereverzin ?

Reply:
Because the scheduled job runs asynchronously - in a background process

Reply:
cool worth putting a comment there